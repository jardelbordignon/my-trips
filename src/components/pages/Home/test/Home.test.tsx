import { render, screen } from '@testing-library/react'

import { Home } from '..'

describe('Home', () => {
  it('deve renderizar um heading com o texto React Avançado', () => {
    render(<Home />)

    expect(
      screen.getByRole('heading', { name: /React Avançado/i })
    ).toBeInTheDocument()
  })

  it('deve renderizar o heading conforme o snapshot', () => {
    const { container } = render(<Home />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('deve renderizar as cores corretamente', () => {
    const { container } = render(<Home />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
