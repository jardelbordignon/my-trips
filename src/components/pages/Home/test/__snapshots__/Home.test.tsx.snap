// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home deve renderizar o heading conforme o snapshot 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 20rem;
  padding: 0 15px;
}

<main
  class="c0"
>
  <h2
    class=""
  >
    My Trips
  </h2>
</main>
`
